zschema/leaves.py

```diff
@@ -72,13 +71,36 @@ def to_es(self):
         self.add_es_var(retv, "analyzer", "es_analyzer", "ES_ANALYZER")
         self.add_es_var(retv, "search_analyzer", "es_search_analyzer",
                 "ES_SEARCH_ANALYZER")
-
         if self.es_include_raw:
             retv["fields"] = {
                     "raw":{"type":"keyword"}
             }
         return retv
 
+    def _docs_common(self, parent_category):
+        retv = {
+            "detail_type": self.__class__.__name__,
+            "category": self.category if self.category else parent_category,
+            "doc": self.doc,
+            "required": self.required,
+        }
+        if hasattr(self, "values_s") and len(self.values_s):
```

Comment:
Why are `retv["values"`] and `retv["examples"]` mutually exclusive?

It would be nice if the values could be individually documented (something like `"Algorithms": Enum(documented_values={"value1": "docs for value1"})`, except more natural)

Reply:
It se ms reasonable to let different types define their own type of '.doc' property. The default could be to just hand back._doc but could also do something else like compilea list of enumerated values into a doc string.

Reply:
> Why are `retv["values"`] and `retv["examples"]` mutually exclusive?

If there is an exhaustive list of possible values, I see no reason to support a list of example values as well.

> It would be nice if the values could be individually documented (something like `"Algorithms": Enum(documented_values={"value1": "docs for value1"})`, except more natural)

Maybe, though I'm unconvinced this is worth the work to implement. The two examples of enums I recall offhand in our schema are elliptic curves and certificate types (leaf/intermediate/root), both of which seem plenty self-explanatory given the existence of a docstring for the field and a list of possible values.

> It se ms reasonable to let different types define their own type of '.doc' property. The default could be to just hand back._doc but could also do something else like compilea list of enumerated values into a doc string.

@zakird I'm not sure I follow what you're suggesting here. As of this PR, types can have a docstring and a list of example values (or a list of possible values, for enums). What are you suggesting should change?